Yosemite/Yosemite/Actions/AppSettingsAction.swift

```diff
@@ -63,4 +63,17 @@ public enum AppSettingsAction: Action {
     /// Clears all the states related to stats version
     ///
     case resetStatsVersionStates
+
+    // MARK: - General App Settings
+
+    /// Saves the `date` as the last known date that the app was installed. This does not do
+    /// anything if there is a persisted installation date already and it is older than the
+    /// given `date`.
+    ///
+    case setInstallationDateIfNecessary(date: Date, onCompletion: ((Result<Void, Error>) -> Void))
```

Comment:
[Newbie Q]: Is the reason on why these methods are asynchronous because the dispatch method don't allow for any return value?

Reply:
I believe so. Also, if I remember correctly, [Redux or Flux](https://facebook.github.io/flux/docs/in-depth-overview) does not even allow return values. The store should always be updated. And the UI should _observe_ the Store's state. Unfortunately, we don't have that architecture right now. It's something that I want to add in the future. 

So, for now, we are left with `onCompletion`. It's a bit cumbersome. 

I do like the concept that this enforces: updating the store should be asynchronous. This means we can improve things in the future so that they will happen in the background.